### PR TITLE
Task/expose image level parameter

### PR DESCRIPTION
**Pull request by canavandl**
ref issue #2040 

This PR exposes a level attr to the Image, ImageURL and ImageRGBA glyphs in order to allow the user to change the default level (from "underlay"), in order to potentially put the image below the grid.

I chose to limit the levels to: image, underlay, and glyph. Images with levels set to overlay or tool seemed to only demonstrated effects that were undesireable (i.e. zooming an image with level='overlay' caused the image to expand over the axes). If anyone can come up with a situation where this behavior is ideal, I can change the PR.

### CODE REVIEW

File: bokeh/models/glyphs.py [hunk 1]
```diff
@@ -346,6 +346,11 @@ def __init__(self, **kwargs):
 
     """)
 
+    level = Enum('image', 'underlay', 'glyph', default='underlay', help="""
+    Specifies the level in which to render the image.
+
```
Comment (damianavila) [inline bokeh/models/glyphs.py]: I think I need some specification about the enum options? Also making underlay the default is not changing the current behaviour?

**Comment by bryevdv** [comment]
@canavandl a few notes:
* The enum type should be defined once in `enums.py` (I think) and should be available on all glyphs, not just the image one. 
* The levels should mirror the JS levels exactly anything else will just lead to confusion

Also in FD I thought the agreement was to look into keeping level on the `GlyphRenderer` and having the plotting methods set it appropriately on the renderer based on a method parameter, e.g. `p.circle(..., level="annotation")` The specific case that worries me with this approach is:
```
circle_glyph.level = "underlay"
circle_renderer.level = "overlay"
```
What happens? Which takes precedence? Any time there is a situation like this with some arbitrary precedence rule it is both hard to remember the behavior, and hard to document/explain exactly what happens in simple terms.

**Comment by bryevdv** [comment]
For some more info about why other levels could be useful: different levels have different clipping behaviours. Everything in "glyph" is clipped to the plot frame, for instance. But "annotation" is not, anything drawing in that level has free range over the canvas. As we start to leverage layout and get more things on one canvas, we might want to have different things drawn *in between* plots. For instance, maybe a trapezoidal "zoom out" patch that connects the small area on the top to the zoomed out area on a plot below. This morel affects explicit annotation objects, but those objects might use glyphs internally for implementation, and want to change the level accordingly. Also just in general, the project history shows that keeping the python and JS more in sync always ends up being the better decision.

**Comment by canavandl** [comment]
> The enum type should be defined once in enums.py (I think) and should be available on all glyphs, not just the image ones

I agree/made the change

>The levels should mirror the JS levels exactly anything else will just lead to confusion

I agree/made the change

> Also in FD I thought the agreement was to look into keeping .level (only) on the GlyphRenderer and having the plotting methods set it appropriately on the renderer based on a method parameter, e.g. p.circle(..., level="annotation") The specific case that worries me with the current approach of delegating is:
circle_glyph.level = "underlay"
circle_renderer.level = "overlay"

I agree/made the change. It doesn't make intuitive sense for a glyph model to have a render level attr, instead that method should be limited to the renderer. The implementation was a lot cleaner too.

>For some more info about why other levels could be useful: different levels have different clipping behaviours. Everything in "glyph" is clipped to the plot frame, for instance. But "annotation" is not, anything drawing in that level has free range over the canvas. As we start to leverage layout and get more things on one canvas, we might want to have different things drawn in between plots. For instance, maybe a trapezoidal "zoom out" patch that connects the small area on the top to the zoomed out area on a plot below. This morel affects explicit annotation objects, but those objects might use glyphs internally for implementation, and want to change the level accordingly. Also just in general, the project history shows that keeping the python and JS more in sync always ends up being the better decision.

Thanks for the longer term insights, I can also imagine wanting to annotate a plot with a text glyph and have that render over the axes.

**Comment by bryevdv** [comment]
LGTM, merging.

**Pull request merged**